packages/tempus/src/Tempus_StepperIMEX_RK_Partition_impl.hpp

```diff
@@ -237,6 +237,8 @@ void StepperIMEX_RK_Partition<Scalar>::setTableaus(
     << "    number of stages = " << explicitTableau_->numStages() << "\n"
     << "  Implicit tableau = " << implicitTableau_->description() << "\n"
     << "    number of stages = " << implicitTableau_->numStages() << "\n");
+
+  this->isInitialized_ = false;
```

Comment:
@rppawlo I am all in favor of simplifying the code/logic.  I had forgot about just allowing a single initialize call, but wouldn't require some code in the set functions to check if isInitialized_ is true and error out?  Otherwise apps could set, initialize and reset, without an error (i.e., a second initialize call is not required)?

I just figured there would be some reason to set, initialize, reset and reinitialize.

Reply:
It would require an assert check on the set calls. My worry is if the user keeps calling initialize after every call to set (have seen teams do this) it could be costly. Does the initialize redo all the consistency checks? Could it keep recomputing a consistent x_dot? The other issue was from the last failed empire merge. The initialize was wiping out some of the info the user set. Hopefully that is fixed with the removal of parameter list from the objects. I like the single initialize. But would anyone want to make changes to the stepper in the middle of a run (e.g. swap out nonlinear solver). That might be a reason to keep this around, but probably not.

Reply:
I have definitely leaned toward having initialize be lightweight. It does not redo any IC consistency work.

Like you said, the problem with the last issue was related to the ParameterLists and reconstructing the Stepper from it.  That has been removed with #5543.

I agree that a single initialize call has its appeal, but I worry that there will be a case come up that needs the reset and reinitialize (e.g., swapping out the solver or changing stepper values).  But on the other side, maybe we should program to current needs and not design for future unspecified needs.  I would lean towards your suggestion, a single initialize.